packages/components/src/custom-select-control/test/index.js

```diff
@@ -82,10 +82,62 @@ const ControlledCustomSelectControl = ( {
 };
 
 describe.each( [
-	[ 'uncontrolled', UncontrolledCustomSelectControl ],
-	[ 'controlled', ControlledCustomSelectControl ],
+	[ 'Uncontrolled', UncontrolledCustomSelectControl ],
+	[ 'Controlled', ControlledCustomSelectControl ],
 ] )( 'CustomSelectControl %s', ( ...modeAndComponent ) => {
-	const [ , Component ] = modeAndComponent;
+	const [ mode, Component ] = modeAndComponent;
+
+	it( 'Should select the first option when no explicit initial value is passed without firing onChange', () => {
+		const mockOnChange = jest.fn();
+		render( <Component { ...props } onChange={ mockOnChange } /> );
+
+		expect(
+			screen.getByRole( 'button', {
+				expanded: false,
+			} )
+		).toHaveTextContent( 'violets' );
+
+		expect( mockOnChange ).not.toHaveBeenCalled();
```

Comment:
No delay / sleep necessary?

This applies to the rest of the places where we're asserting whether `onChange` callback was called.

Reply:
In my tests, it was not necessary for the v1 version, while it was unfortunately necessary for the ariakit-based version